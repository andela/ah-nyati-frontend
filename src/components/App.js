import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Notifications from 'react-notify-toast';
import { ToastContainer } from 'react-toastify';
import jwtDecode from 'jwt-decode';
import ReduxToastr from 'react-redux-toastr';
import NavBar from './NavigationBar';
import Homepage from '../views/Homepage';
import Register from '../views/Register';
import Login from './LoginForm/Index';
import About from '../views/about';
import AuthUserProfile from '../views/Dashboard';
import NotFound from '../views/NotFound';
import SocialAuth from '../views/SocialAuth';
import ResetPassword from '../views/ResetPassword';
import PasswordReset from './PasswordReset/PasswordReset';
import Articles from '../views/Articles';
import Profile from '../views/Profile';
import store from '../store';
import setAuthToken from '../utils/setAuthToken';
import { setCurrentUser } from '../actions/authActions';
import CreateArticlePage from './article/CreateArticlePage';
import UpdateArticlePage from './article/UpdateArticlePage';
import SingleArticle from './SingleArticle';
import Comments from './Comments';

const token = localStorage.jwtToken;
let loggedInUser = {};
/* istanbul ignore next */
if (token) {
  setAuthToken(token);
  const decoded = jwtDecode(token);
  loggedInUser = store.dispatch(setCurrentUser(decoded));
}


const App = () => (
  <React.Fragment>
    <ReduxToastr
      timeOut={4000}
      newestOnTop={false}
      preventDuplicates
      position="top-right"
      transitionIn="fadeIn"
      transitionOut="fadeOut"
      progressBar
      closeOnToastrClick
    />
    <Notifications />
    <NavBar user={loggedInUser} />
    <ToastContainer />
    <Switch>
      <Route exact path="/" component={Homepage} />
      <Route path="/register" component={Register} />
      <Route path="/authUserProfile" component={AuthUserProfile} />
      <Route path="/login" component={Login} />
      <Route path="/about" component={About} />
      <Route path="/resetPassword" component={ResetPassword} />
      <Route path="/newpassword" component={PasswordReset} />
      <Route path="/socialAuth" component={SocialAuth} />
      <Route path="/article" component={CreateArticlePage} />
      <Route exact path="/articles" component={Articles} />
      <Route path="/articles/:slug" component={SingleArticle} />
      <Route path="/comments/" component={Comments} />
      <Route path="/updatearticle/:slug" component={UpdateArticlePage} />
      <Route path="/profile/:userId" component={Profile} />

      <Route component={NotFound} />
    </Switch>
  </React.Fragment>
);

export default App;
