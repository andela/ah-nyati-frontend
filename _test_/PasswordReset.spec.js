import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';

import App from '../src/components/PasswordReset/PasswordReset';

describe('App', () => {
  let app;
  beforeEach(() => {
    app = shallow(<App />);
  });

  it('renders successfully', () => {
    expect(app).toBeDefined();
  });

  it('renders a div tag', () => {
    expect(app.find('div').length).toBe(2);
  });

  it('renders a form tag', () => {
    expect(app.find('form').length).toBe(1);
  });

  it('renders a p tag', () => {
    expect(app.find('p').length).toBe(2);
  });

  it('renders a input tag', () => {
    expect(app.find('input').length).toBe(3);
  });
});
