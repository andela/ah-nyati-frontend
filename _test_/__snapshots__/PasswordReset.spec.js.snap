// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Simulates a event on form password input 1`] = `ShallowWrapper {}`;

exports[`App renders snapshot uccessfully 1`] = `ShallowWrapper {}`;
